### ISSUE TITLE
Futures Trading

### ISSUE THREAD

**Issue by wycfutures** [microsoft/qlib#107]
## 🌟 Feature Description
Futures trading support

Do you have any plans to add the futures trading support to the framework?

**Comment by you-n-g** [comment]
Yes.  Qlib is designed to support all financial derivatives include futures. 
But futures trading is quite a broad and somewhat fuzzy word.   Can you give some examples of your future trading research works and what features are required to support your research?

**Comment by nkiki** [comment]
> Yes. Qlib is designed to support all financial derivatives include futures.
> But futures trading is quite a broad and somewhat fuzzy word. Can you give some examples of your future trading research works and what features are required to support your research?

how about cryptocurrency?:)

**Comment by you-n-g** [comment]
@nkiki  I think most of the research works on cryptocurrency are similar to  alpha forecasting and High-frequency trading .
For these two cases, the former has been alreadly supported by Qlib's framework areadly (But the data is not included.)
The latter will be supported in the future.

**Comment by wycfutures** [comment]
> Yes. Qlib is designed to support all financial derivatives include futures.
> But futures trading is quite a broad and somewhat fuzzy word. Can you give some examples of your future trading research works and what features are required to support your research?

Like reinforcement learning based "Medium-frequency" trading strategies

**Comment by you-n-g** [comment]
@wycfutures  The  RL diagram is not supported in the current version, but it is what we are designing and we will try to implement a framework in the future.  
Welcome to discuss with us about the design of framework.

**Comment by johnputmanii** [comment]
I'd be happy to see the existing framework support the S&P 500 E mini: https://finance.yahoo.com/quote/ES=F/

**Comment by ChengYen-Tang** [comment]
> @nkiki I think most of the research works on cryptocurrency are similar to alpha forecasting and High-frequency trading .
> For these two cases, the former has been alreadly supported by Qlib's framework areadly (But the data is not included.)
> The latter will be supported in the future.

Hi @you-n-g ,
Are there any plans to add the cryptocurrency trading environment in the backtest?
For example: Okex Perpetual Swap ?

**Comment by you-n-g** [comment]
@ChengYen-Tang Currently, we are working on developing high-frequency trading based on supervised learning.
A framework and some simple examples will be included.
You can check if your requriements can be fulfilled.

**Comment by ChengYen-Tang** [comment]
Hi @you-n-g ,
I have a question.
Does the backtest include short-term operations?
Also, are there relevant documents that include the backtest environment?

**Comment by you-n-g** [comment]
Hi, @ChengYen-Tang 
Here is some documents about backtest environment (some interfaces may be refactored recently).
https://qlib.readthedocs.io/en/latest/component/backtest.html

Short-term operation may be supported in the next version

**Comment by ChengYen-Tang** [comment]
OK, Hope this problem can be resolved as soon as possible, thank you.
https://github.com/microsoft/qlib/issues/237
